server/channels/store/sqlstore/user_store.go

```diff
@@ -1613,6 +1613,8 @@ func generateSearchQuery(query sq.SelectBuilder, terms []string, fields []string
 			}
 			termArgs = append(termArgs, fmt.Sprintf("%%%s%%", strings.TrimLeft(term, "@")))
 		}
+		searchFields = append(searchFields, "Id = ?")
```

Comment:
Not sure if it is going to make things any better, but we could add this search field only if the term "looks like a user id" (specially since we are doing only exact match, and not partial match)

My idea is that a regex should be cheaper that one more comparison on each user. But maybe the database is doing that in a smart way where it does not make real difference.

Reply:
Another way to look at this would be "if the term looks like an id, get directly the user from the database, and if it doesn't exist, do the search".

I would say 99% of the searches will not "look like an id" unless they are indeed an id, so we will often "guess right" and do a cheaper operation (get user by id) than doing the whole search.

0/5

Reply:
And one last comment about this...

If a user has a name that is the same as the ID of another user (not sure if the length of the name fields allow it though), your solution would show both the user and the user with such a weird name, right? My proposed solution would only show the user with that ID. Not sure if that is an issue.